chart/templates/patch_serviceaccount.yaml

```diff
@@ -0,0 +1,66 @@
+{{- include "_config.load" $ }}
+{{- if .Values.kube.image_pull_secrets }}
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  annotations:
+    "helm.sh/hook": pre-install
+    "helm.sh/hook-weight": "-5"
+    "helm.sh/hook-delete-policy": hook-succeeded
+  name: patch-sa
+  namespace: {{ .Release.Namespace | quote }}
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - serviceaccounts
+  verbs:
+  - get
+  - patch
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  annotations:
+    "helm.sh/hook": pre-install
+    "helm.sh/hook-weight": "-5"
+    "helm.sh/hook-delete-policy": hook-succeeded
+  name: patch-sa
+  namespace: {{ .Release.Namespace | quote }}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: patch-sa
+subjects:
+- kind: ServiceAccount
+  name: default
+  namespace: {{ .Release.Namespace | quote }}
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: patch-sa
+  namespace: {{ .Release.Namespace | quote }}
+  annotations:
+    "helm.sh/hook": pre-install
+    "helm.sh/hook-delete-policy": hook-succeeded
+spec:
+  template:
+    spec:
+      restartPolicy: Never
+      containers:
+      - name: sa
+        {{- with $image := $.Values.releases.kubecf_kubectl.image }}
+        image: {{ printf "%s:%s" $image.repository $image.tag | quote }}
+        {{- end }}
+        imagePullPolicy: IfNotPresent
+# set image_pull_secrets list
+{{- $image_pull_secret_names := list }}
+{{- range $secret_name := $.Values.kube.image_pull_secrets }}
+  {{- $image_pull_secret_names = append $image_pull_secret_names (dict "name" $secret_name) }}
+{{- end }}
+        command: ["/bin/sh", "-c"]
+        args:
+        - |
+          kubectl patch serviceaccount default -p '{"imagePullSecrets": {{ $image_pull_secret_names | toJson }} }' -n {{ .Release.Namespace }}
```

Comment:
I've just been wondering what happens if the service account already has `imagePullSecrets` defined: they will just get overwritten.

A quick search doesn't find any good way to append to a list. I found [this](https://stackoverflow.com/a/63438403/284266) on StackOverflow, but I'm not sure if this works when the list is empty initially.

I think it is fine to not do anything about this, but maybe add a comment that any existing secrets on the SA will also have to be specified again via the Helm setting.

Reply:
If there's already an imagepullsecret within a serviceaccount, a patch just adds the rest in.

Example:
```
$ kubectl describe sa -n cfmr default
Name:                default
Namespace:           cfmr
Labels:              <none>
Annotations:         <none>
Image pull secrets:  default-dockercfg-sww5b # <~ Default imagepullsecret openshift adds for its private image registry
Mountable secrets:   default-token-fd55q
                     default-dockercfg-sww5b
Tokens:              default-token-fd55q
                     default-token-lrmfs
Events:              <none>

$ kubectl patch serviceaccount default -p '{"imagePullSecrets": [ {"name":"foo"}, {"name":"bar"} ] }' -n cfmr
serviceaccount/default patched

$ kubectl describe sa -n cfmr default
Name:                default
Namespace:           cfmr
Labels:              <none>
Annotations:         <none>
Image pull secrets:  foo (not found)
                     bar (not found)
                     default-dockercfg-sww5b
Mountable secrets:   default-token-fd55q
                     default-dockercfg-sww5b
Tokens:              default-token-fd55q
                     default-token-lrmfs
Events:              <none>
```

Reply:
confirmed and comments added